### PR TITLE
Chemfiles updates

### LINKED ISSUE #3216: Add CI runner with older versions of chemfiles
> Can you open an issue about covering this so we don't forget? I was working on a weekly cron CI job somewhere else, I can probably just add a workflow that installs older versions of chemfiles.

_Originally posted by @IAlibay in https://github.com/MDAnalysis/mdanalysis/pull/3126#r607320587_

### LINKED ISSUE #3492: DeprecationWarning: deprecated distutils Version classes
This is popping up in CI - there's at least two uses of LooseVersion in our codebase that probably needs fixing (chemfiles.py and setup.py).

```
/home/runner/work/mdanalysis/mdanalysis/package/MDAnalysis/coordinates/chemfiles.py:108: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  MIN_CHEMFILES_VERSION = LooseVersion("0.9")
```

### LINKED ISSUE #3498: Drop support for chemfiles < 0.10
Related to #3490 

Given that we now only support py3.7+ it would make sense to just drop support for old versions of chemfiles and simplify our code paths.

### PR DESCRIPTION
**Pull request by Luthaf**
Fixes #3492
Fixes #3498
Fixes #3216 

Changes made in this Pull Request:
 - Use `packaging.version.Version` instead of deprecated `distutils.version.LooseVersion`
- Drop support for chemfiles 0.9

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/coordinates/chemfiles.py [hunk 1]
```diff
@@ -126,7 +125,7 @@ def check_chemfiles_version():
             "Try installing with 'pip install chemfiles'"
         )
         return False
-    version = LooseVersion(chemfiles.__version__)
+    version = Version(chemfiles.__version__)
     wrong = version < MIN_CHEMFILES_VERSION or version >= MAX_CHEMFILES_VERSION
     if wrong:
         warnings.warn(
```
Comment (IAlibay) [inline package/MDAnalysis/coordinates/chemfiles.py]: Why do we have both a warning and a subsequent error doing the same thing? isn't it easier/cleaner to just do the error?

File: testsuite/MDAnalysisTests/coordinates/test_chemfiles.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def test_version_check(self):
         import chemfiles
 
         actual_version = chemfiles.__version__
-        chemfiles.__version__ = "0.8.3"
+        chemfiles.__version__ = "0.9.3"
         assert not check_chemfiles_version()
 
         with pytest.raises(RuntimeError, match="Please install Chemfiles > 0.9"):
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/coordinates/test_chemfiles.py]: matches need updating here?

**Comment by Luthaf** [comment]
CI errors seems unrelated, they are exception from the `parse_latex` function of the `citeproc` module

**Review by IAlibay — CHANGES_REQUESTED** [review]
Thanks for working on this @Luthaf.

Can you elaborate on where you are seeing these CI errors? I'm seeing this in the tests "RuntimeError: Please install Chemfiles > 0.10" which seems like somehow the versions aren't getting checked properly?

File: testsuite/MDAnalysisTests/coordinates/test_chemfiles.py (hunk 2 above)
Comment (IAlibay) [reply testsuite/MDAnalysisTests/coordinates/test_chemfiles.py]: Ok the failures are specifically due to this test. The test fails early on the first `pytest.raises` so the version doesn't get set back to the correct value.

I have put a PR against your feature branch with a monkeypatch rewrite that should avoid this issue.

File: package/MDAnalysis/coordinates/chemfiles.py (hunk 1 above)
Comment (Luthaf) [reply package/MDAnalysis/coordinates/chemfiles.py]: I think the idea was to be able to send a warning on import (and error on usage), but it is not implemented. I'll change it to only error instead

**Comment by Luthaf** [comment]
> Can you elaborate on where you are seeing these CI errors?

My bad, I clicked on a random failed runner and got what looked like unrelated errors (https://github.com/MDAnalysis/mdanalysis/runs/4727027488?check_suite_focus=true).

**Review by IAlibay — APPROVED** [review]
Thanks @Luthaf !

**Pull request merged**